### DISCUSSION TITLE
How to add a mono-repo config, test it and add it to public renovate presets

### DISCUSSION THREAD

**Discussion by sommmen** [renovatebot/renovate#14987] [Request Help] [answered]
Hiya,

My current renovate.json after onboarding looks  like this, i'm mainly using nuget/.NET with renovate.

``` csharp
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json"
}
```

I have the following monorepo that is giving me problems now because they're not applied as a single change.

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet

I have kind of a 3 part question regarding this:
- How would i add a mono-repo to my renovate.json config file?
- How would i be able to test this locally without opening a bunch of prs and so on?
- How would i add a mono-repo to the presets of renovate? Whats the process here?

**Comment by sommmen** [comment]
Regarding my third point, i think i came across where i could publish monorepo presets here:

https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/monorepo.ts

**Comment by rarkins** [comment, marked as answer]
1. Define a package rule with matching patterns and a `groupName`
2. Easiest is to use a test repo
